inference-engine/tests/functional/plugin/cpu/test_utils/cpu_test_utils.hpp

```diff
@@ -0,0 +1,149 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "functional_test_utils/layer_test_utils.hpp"
+#include "ie_system_conf.h"
+
+#include <ngraph/variant.hpp>
+#include <exec_graph_info.hpp>
+
+using namespace InferenceEngine;
+
+namespace CPUTestUtils {
+
+typedef enum {
+    nchw,
+    nChw8c,
+    nChw16c,
+    ncdhw,
+    nCdhw8c,
+    nCdhw16c,
+    goihw,
+    undef
+} cpu_memory_format_t;
+
+typedef std::tuple<
+        std::vector<cpu_memory_format_t>,
+        std::vector<cpu_memory_format_t>,
+        std::vector<std::string>,
+        std::string> CPUSpecificParams;
+
+namespace GroupConv {
+
+/* CPU PARAMS */
+const auto cpuParams_ref_2D = CPUSpecificParams{{nchw}, {nchw}, {"ref_any"}, "ref_any_FP32"};
+const auto cpuParams_ref_3D = CPUSpecificParams{{ncdhw}, {ncdhw}, {"ref_any"}, "ref_any_FP32"};
+
+const auto cpuParams_gemm_2D = CPUSpecificParams{{nchw}, {nchw}, {"gemm_any"}, "jit_gemm_FP32"};
+const auto cpuParams_gemm_3D = CPUSpecificParams{{ncdhw}, {ncdhw}, {"gemm_any"}, "jit_gemm_FP32"};
+
+const auto cpuParams_sse42_2D = CPUSpecificParams{{nChw8c}, {nChw8c}, {"jit_sse42"}, "jit_sse42_FP32"};
+const auto cpuParams_sse42_3D = CPUSpecificParams{{nCdhw8c}, {nCdhw8c}, {"jit_sse42"}, "jit_sse42_FP32"};
+const auto cpuParams_sse42_dw_2D = CPUSpecificParams{{nChw8c}, {nChw8c}, {"jit_sse42_dw"}, "jit_sse42_dw_FP32"};
+const auto cpuParams_sse42_dw_3D = CPUSpecificParams{{nCdhw8c}, {nCdhw8c}, {"jit_sse42_dw"}, "jit_sse42_dw_FP32"};
+
+const auto cpuParams_avx2_2D = CPUSpecificParams{{nChw8c}, {nChw8c}, {"jit_avx2"}, "jit_avx2_FP32"};
+const auto cpuParams_avx2_3D = CPUSpecificParams{{nCdhw8c}, {nCdhw8c}, {"jit_avx2"}, "jit_avx2_FP32"};
+const auto cpuParams_avx2_dw_2D = CPUSpecificParams{{nChw8c}, {nChw8c}, {"jit_avx2_dw"}, "jit_avx2_dw_FP32"};
+const auto cpuParams_avx2_dw_3D = CPUSpecificParams{{nCdhw8c}, {nCdhw8c}, {"jit_avx2_dw"}, "jit_avx2_dw_FP32"};
+
+const auto cpuParams_avx512_2D = CPUSpecificParams{{nChw16c}, {nChw16c}, {"jit_avx512"}, "jit_avx512_FP32"};
+const auto cpuParams_avx512_3D = CPUSpecificParams{{nCdhw16c}, {nCdhw16c}, {"jit_avx512"}, "jit_avx512_FP32"};
+const auto cpuParams_avx512_dw_2D = CPUSpecificParams{{nChw16c}, {nChw16c}, {"jit_avx512_dw"}, "jit_avx512_dw_FP32"};
+const auto cpuParams_avx512_dw_3D = CPUSpecificParams{{nCdhw16c}, {nCdhw16c}, {"jit_avx512_dw"}, "jit_avx512_dw_FP32"};
```

Comment:
Looks like it might be good idea to put layout in the name. For example for some layers I will need to check nchw precision for avx2 ISA.